awsproviderlint/passes/AWSAT003/AWSAT003.go

```diff
@@ -0,0 +1,73 @@
+// Package AWSAT003 defines an Analyzer that checks for
+// hardcoded regions
+package AWSAT003
+
+import (
+	"go/ast"
+	"go/token"
+	"regexp"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/bflad/tfproviderlint/passes/commentignore"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+const Doc = `check for hardcoded regions
+
+The AWSAT003 analyzer reports hardcoded regions. Testing in non-standard
+partitions with hardcoded regions (and AZs) will cause the tests to fail. 
+`
+
+const analyzerName = "AWSAT003"
+
+var Analyzer = &analysis.Analyzer{
+	Name: analyzerName,
+	Doc:  Doc,
+	Requires: []*analysis.Analyzer{
+		commentignore.Analyzer,
+		inspect.Analyzer,
+	},
+	Run: run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	ignorer := pass.ResultOf[commentignore.Analyzer].(*commentignore.Ignorer)
+	inspect := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+
+	nodeFilter := []ast.Node{
+		(*ast.BasicLit)(nil),
+	}
+
+	resolver := endpoints.DefaultResolver()
+	partitions := resolver.(endpoints.EnumPartitions).Partitions()
+	var regions []string
+
+	for _, p := range partitions {
+		for id, _ := range p.Regions() {
+			regions = append(regions, id)
+		}
+	}
+
+	re := regexp.MustCompile(strings.Join(regions, "|"))
```

Comment:
Ha here's a (relatively minor/fun) false positive:

https://github.com/terraform-providers/terraform-provider-aws/blob/b77516e01695792da30c58b73924e79c753180df/aws/resource_aws_codebuild_project_test.go#L3460

I think the only way to workaround that would be to look for double quotes or something:

```go
regexp.MustCompile(fmt.Sprintf(`"(%s)[a-z]?", strings.Join(regions, "|")))
```

But I'm guessing that could also lead to it missing some things, like when a region/AZ is hardcoded in the string for a TestCheck. 🤔  Not a big deal, just noting.

Reply:
I figure there will be a few places that need to use ignore.